src/main/java/io/boomerang/eventing/nats/jetstream/SubjectMatchChecker.java

```diff
@@ -0,0 +1,93 @@
+package io.boomerang.eventing.nats.jetstream;
+
+import java.util.Arrays;
+
+class SubjectMatchChecker {
```

Comment:
What is the purpose of this class?

Reply:
I don't understand why the message subject and the Streams subject have to be tightly coupled? I didn't think we related the two in Listener?

Reply:
Hey @corcoja just checking in on this

Reply:
@tlawrie, the message subject and the Streams subject is tightly coupled for a single `TinCanCommunication` object. If you want to work with multiple Streams with various subjects, then you just create multiple `TinCanCommunication` objects.

In other words, a `TinCanCommunication` works around just on subject (can be a wildcard): you can publish and listen for messages with a specific subject.

In your project, you can do the following (pseudocode):

```
x = TinCanCommunication(Stream("some.subject.>"), Consumer());
x.publish("some.subject.test", "A test message!"); // Will publish a message with subject "some.subject.test"
x.subscribe(); // Will receive the message "A test message!"
```

And if you want to cover multiple subjects in a project, you can create more than one `TinCanCommunication`.

With the above example, if I don't check for the published subject to match streams subject, the `subscribe` method wouldn't return anything. Under the hood, publishing a message to either a Jetstream or to a NATS queue, it is the same CLI command:

```bash
nats pub some.subject.test "A test message!"
```

If you have a stream that matches subject "some.subject.test", the message will pe published to that stream, otherwise, it goes to NATS queue.

Reply:
Awesome. Ok. So when we say tightly coupled subject, its between the Publisher and Subscriber. Not the CloudEvent subject. Is that right?

Reply:
Yup, that's right! You can have as much CloudEvent subjects, as you want. Each one managed individually.

But that's just one scenario, you can have even more "advanced" stuff, like referring to the same Stream from multiple `TinCanCommunication` objects and having Consumers with filters. E. g.: (pseudocode):

```
a = TinCanCommunication(Stream("payment.>"), Consumer()); // Consumer doesn't really matter here as we just publish
a.publish("payment.pending", "Your payment is pending...");
a.publish("payment.status.update", "Your payment is being processed...");
a.publish("payment.status.update", "Your payment has been accepted!");

b = TinCanCommunication(Stream("payment.*"), Consumer(filter="payment.pending")); // Points to the same Stream as above but the Consumer will filter just the messages with subject "payment.pending"
b.subscribe(); // Will receive the message "Your payment is pending..."

c = TinCanCommunication(Stream("payment.*"), Consumer(filter="payment.status.update")); // Again, points to the same Stream but this time, Consumer filters just the messages with subject "payment.status.update"
c.subscribe(); // Will receive the messages "Your payment is being processed..." and "Your payment has been accepted!"
```